### ISSUE TITLE
Missing metric, and log error `One or more TimeSeries could not be written: Unrecognized region or location`

### ISSUE THREAD

**Issue by willchan** [GoogleCloudPlatform/prometheus-engine#153]
I have a GKE cluster with self-installed managed collection, using https://raw.githubusercontent.com/GoogleCloudPlatform/prometheus-engine/v0.3.0/manifests/setup.yaml and https://raw.githubusercontent.com/GoogleCloudPlatform/prometheus-engine/v0.3.0/manifests/operator.yaml

I've added PodMonitoring custom resources for many of our pods. Most of them seem to work, but I see at least one that is missing. It's unclear to me why it doesn't show up.

In the logs, the main error I see are log entries of this form. Of course, I can't tell what metric this is referring to. I don't know where the `Unrecognized region or location` is from, but I do note that the collector is running with `--export.label.location=us-west1-a`, which I assume the operator auto-detected in GKE.

`ts=2022-03-04T01:43:50.939Z caller=log.go:124 component=gcm_exporter level=error msg="send batch" size=200 err="rpc error: code = InvalidArgument desc = One or more TimeSeries could not be written: Unrecognized region or location.: timeSeries[1,2,4-17,19-21,23-37,40-48,50,52-58,60,61,63-90,92-102,104-107,110-112,114-117,119,120,122-124,126-132,134-142,144-151,154-188,190-197,199]`

Also, the metric that is missing looks like:

```
# TYPE sorter_qc_counts gauge
sorter_qc_counts{batch="2711",lane="1",location="BOTTOM",rack="70",vision_label="M"} 2344.0
```

The labelset for this gauge is substantial, on the order of 300 combinations. I dunno if that could be related.

**Comment by lyanco** [comment]
Hi Will,

`location` is a reserved label in our system (along with project_id, cluster, namespace, job, and instance). 

What's happening here is your `location` label is accidentally overwriting the reserved `location` label, making our API think that you intend to write to a GCP region called `BOTTOM`, which of course doesn't exist.

We want to preserve this behavior, because there are some valid use cases for overwriting `location` on the metric. A canonical example is, you're a EU operator generating metrics from EU-owned cell phones, and you want to make sure their data is stored in the EU even if the cell phone is currently in the US, so you would manually override `location` to ensure the data gets saved in the EU.

The easiest solution here would be to just use a different label name. Would that be acceptable?

Regardless we should document this a bit more clearly.

**Comment by willchan** [comment]
Aha, that's useful to know. Thanks for the quick reply.

Do we need to fix this in application code, or can we relabel with a RelabelingRule to address this?

I don't know if it's easy, but adding error handling for the conflict would be nice, so we could get a better log message without having to bug y'all. But your response time is very good, so much appreciated :)

**Comment by lyanco** [comment]
I think the relabeling rule will work, but I am not positive. Our system will auto-fill in any missing `project_id` or `location` labels before it's sent to the API. I think that happens after relabeling rules are applied but I'm not positive. Should be pretty simple for you to try out, though.

**Comment by willchan** [comment]
OK cool, thanks. I'll give the relabeling a try first. Multiple applications use this same label and we alert on these labelsets, so it's a more difficult to update all the binaries.

Also, is there any documentation on the functionality of these reserved labels? We have k8s clusters outside of GKE on which I'd like to deploy the collector. It's unclear to me what location I should pick when configuring the operator, and if it matters.

**Comment by lyanco** [comment]
There is some documentation here: https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-unmanaged#gmp-run-binary

We recommend picking the closest GCP region to wherever your k8s cluster is running. project_id just needs to be a valid GCP project. cluster/namespace/job/location should be automatically added.

**Comment by willchan** [comment]
I've tested out using a [RelabelingRule](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/v0.3.0/doc/api.md#relabelingrule) to replace the label, but it doesn't seem to work. We have numerous applications that use this label, so it's burdensome to fix each of them. Is `location` required for any functionality? If not, would it be possible to remove this requirement that it not be set by applications? The docs don't explain what the values used for, and if it's not providing value, it'd be nicer for migrating users not to have to worry about relabeling their metrics. Alternatively, could the team rework it so a RelabelingRule can solve this rather than having to update all applications that use this label?

**Comment by lyanco** [comment]
It's used to select the GCP region where your data is stored, so it actually has critical functionality.

Can you post your relabeling rule so I can take a look?

Special case handling for relabeling metric-set `location` labels would make sense.

**Comment by lyanco** [comment]
OK, I've created an internal bug for this. I'll update you when complete. However, to be frank, It will probably be some time.

**Comment by paulleroyza** [comment]
Since this is google or GCP specific labels that will overlap many use cases, doesn't it make sense to prefix these like any other system label like `google_` or `gcp_` (or `gc_`) or `prometheus_` or `nodejs_`? I find it useful that I can inject this this into the monitoring stream though.

**Comment by lyanco** [comment]
This is fixed in 0.4.3 which is not released in all GKE clusters.

If you use `location` it will be automatically relabeled to `exported_location` and won't throw an error any more.
